src/completion.rs

```diff
@@ -0,0 +1,2075 @@
+                "from {}",
+                self.package
+            ))),
+            filter_text: Some(self.name.clone()),
+            insert_text: Some(self.name.clone()),
+            insert_text_format: Some(
+                lsp::InsertTextFormat::PlainText,
+            ),
+            kind: Some(lsp::CompletionItemKind::Variable),
+            preselect: None,
+            sort_text: Some(format!(
+                "{} {}",
+                self.name, self.package
+            )),
+            text_edit: None,
+            command: None,
+            data: None,
+            insert_text_mode: None,
+            tags: None,
+        }
+    }
+
+    fn matches(&self, text: String, info: CompletionInfo) -> bool {
+        let imports = info.imports;
+        if self.package == PRELUDE_PACKAGE
+            && fuzzy_match(self.name.as_str(), text.as_str())
+        {
+            return true;
+        }
+
+        if !imports.into_iter().any(|x| self.package == x.path) {
+            return false;
+        }
+
+        if text.ends_with('.') {
+            let mtext = text[..text.len() - 1].to_string();
+            return Some(mtext) == self.package_name;
+        }
+
+        false
+    }
+}
+
+impl CompletionVarResult {
+    fn detail(&self) -> String {
+        match self.var_type {
+            CompletionVarType::Array => "Array".to_string(),
+            CompletionVarType::Bool => "Boolean".to_string(),
+            CompletionVarType::Duration => "Duration".to_string(),
+            CompletionVarType::Float => "Float".to_string(),
+            CompletionVarType::Int => "Integer".to_string(),
+            CompletionVarType::Object => "Object".to_string(),
+            CompletionVarType::Regexp => {
+                "Regular Expression".to_string()
+            }
+            CompletionVarType::String => "String".to_string(),
+            CompletionVarType::Record => "Record".to_string(),
+            CompletionVarType::Time => "Time".to_string(),
+            CompletionVarType::Uint => "Unsigned Integer".to_string(),
+        }
+    }
+}
+
+// fn create_diagnostics_notification(
+//     uri: lsp::Url,
+//     diagnostics: Vec<lsp::Diagnostic>,
+// ) -> Notification<lsp::PublishDiagnosticsParams> {
+//     let method = String::from("textDocument/publishDiagnostics");
+//     let params = lsp::PublishDiagnosticsParams {
+//         uri,
+//         diagnostics,
+//         version: None,
+//     };
+//     Notification { method, params }
+// }
+// #[derive(Serialize, Deserialize)]
+// struct Notification<T> {
+//     method: String,
+//     params: T,
+// }
+//
+// impl<T> Notification<T>
+// where
+//     T: Serialize,
+// {
+//     fn to_json(&self) -> Result<String, String> {
+//         match serde_json::to_string(self) {
+//             Ok(s) => Ok(s),
+//             Err(_) => Err(String::from(
+//                 "Failed to serialize initialize response",
+//             )),
+//         }
+//     }
+// }
```

Comment:
Usually with commented out code, I suggest removing it, especially if there will be a copy in source control that could be linked from an issue. At the very least I would expect a comment providing some context here.